amundsen_gremlin/gremlin_model.py

```diff
@@ -0,0 +1,492 @@
+                           MagicProperties.TO.value, WellKnownProperties.Created.value})
+
+        # NB: Some edge types may not make sense to soft-expire
+        if expirable:
+            properties.update({WellKnownProperties.Expired.value})
+            id_format = kwargs.pop('id_format', EdgeTypeIdFormats.EXPIRABLE.value)
+        else:
+            id_format = kwargs.pop('id_format', EdgeTypeIdFormats.DEFAULT.value)
+
+        parameters = _discover_parameters(id_format)
+        properties_names = set([p.name for p in properties])
+        assert all(p in properties_names for p in parameters), \
+            f'id_format: {id_format} has parameters: {parameters} not found in our properties {properties_names}'
+
+        return cls(properties=tuple(properties), id_format=id_format, **kwargs)
+
+    @lru_cache()
+    def properties_as_map(self) -> Mapping[str, Property]:
+        mapping = dict([(p.name, p) for p in self.properties])
+        assert len(mapping) == len(self.properties), f'are property names not unique? {self.properties}'
+        return mapping
+
+    def id(self, **entity: Any) -> str:
+        # format them if they're not already.  (the isinstance(v, str) feels wrong here tho)
+        values = dict([(n, (self.properties_as_map()[n].format(v) if v is not None and not isinstance(v, str) else v))
+                       for n, v in entity.items()])
+        values.update({'~label': self.label})
+        return self.id_format.format(**values)
+
+    def create(self, **properties: Any) -> Mapping[str, Any]:
+        properties = dict(properties)
+        if MagicProperties.ID.value in self.properties and MagicProperties.ID.value.name not in properties:
+            properties[MagicProperties.ID.value.name] = self.id(**properties)
+        if MagicProperties.LABEL.value in self.properties and MagicProperties.LABEL.value.name not in properties:
+            properties[MagicProperties.LABEL.value.name] = self.label
+        # remove missing values
+        for k in [k for k, v in properties.items() if v is None]:
+            del properties[k]
+        properties.update([(k, v) for k, v in self.properties_as_map().items()
+                           if v.default is not None and k not in properties])
+        property_names = set(self.properties_as_map().keys())
+        assert set(properties.keys()).issubset(property_names), \
+            f'unexpected properties: properties: {properties}, expected names: {property_names}'
+        required_property_names = set([k for k, v in self.properties_as_map().items() if v.required])
+        assert set(properties.keys()).issuperset(required_property_names), \
+            f'expected required properties: properties: {properties}, expected names: {required_property_names}'
+        return properties
+
+
+class VertexTypes(Enum):
+    """
+    In general, you will need to reload all your data: 1. change label, 2. if you change the type of a property,
```

Comment:
I think this comment is missing some words that make it easier to parse.
```suggestion
    In general, you will need to reload all your data if you: 1. change label, 2. if you change the type of a property,
```